edsnlp/pipes/ner/suicide_attempt/suicide_attempt.py

```diff
@@ -0,0 +1,175 @@
+from edsnlp.matchers.regex import RegexMatcher
+from edsnlp.matchers.utils import Patterns
+from edsnlp.pipes.base import BaseNERComponent
+from edsnlp.utils.span_getters import SpanSetterArg
+
+from .patterns import patterns
+
+
+class SuicideAttemptMatcher(BaseNERComponent):
+    """
+    The `eds.suicide_attempt` pipeline component detects mentions of Suicide Attempt.
+    It can be used with a span qualifier
+    for contextualisation of the entity (history) and to detect false positives as
+    negation, hypothesis or family. We recomend to use a Machine Learning qualifier
+    to disambiguate polysemic words, as proposed in [@bey_natural_2024].
+
+    It was designed at AP-HP's EDS, following the insights of the algorithm proposed
+    by [@bey_natural_2024].
+
+    It sets the entity label `suicide_attempt`.
+
+    Extensions
+    ----------
+    Each entity span will have the suicide attempt modality as an attribute.
+    The available modalities are:
+
+        - suicide_attempt_unspecific
+        - autolysis
+        - intentional_drug_overdose
+        - jumping_from_height
+        - cuts
+        - strangling
+        - self_destructive_behavior
+        - burn_gas_caustic
+
+    Examples
+    --------
+    ```python
+    import edsnlp, edsnlp.pipes as eds
+
+    nlp = edsnlp.blank("eds")
+    nlp.add_pipe(eds.suicide_attempt())
+
+    text = "J'ai vu le patient à cause d'une IMV avec paracetamol"
+    doc = nlp(text)
+    doc.ents
+    # Out: (IMV,)
+
+    ent = doc.ents[0]
+    ent._.suicide_attempt_modality
+    # Out: 'intentional_drug_overdose'
+    ```
+
+    ??? info "Patterns used for the NER"
+        ```{ .python .no-check }
+        # fmt: off
+        --8<-- "edsnlp/pipes/ner/suicide_attempt/patterns.py"
+        # fmt: on
+        ```
+
+
+    Parameters
+    ----------
+    nlp : PipelineProtocol
+        The pipeline object.
+    name : str
+        The name of the pipe
+    attr : Union[str, Dict[str, str]]
+        Attribute to match on, eg `TEXT`, `NORM`, etc.
+    ignore_excluded : bool
+        Whether to skip excluded tokens during matching.
+    ignore_space_tokens : bool
+        Whether to skip space tokens during matching.
+    patterns : List[str]
+        The regex pattern to use
+    span_setter : SpanSetterArg
+        How to set matches on the doc
+    label : str
+        Label name to use for the `Span` object and the extension
+
+    Returns
+    -------
+    SuicideAttemptMatcher
+
+    Authors and citation
+    --------------------
+    The `eds.suicide_attempt` component was developed by AP-HP's Data Science
+    team, following the insights of the algorithm proposed
+    by [@bey_natural_2024].
+    """
+
+    def __init__(
+        self,
+        nlp: PipelineProtocol,
+        label: str = "suicide_attempt",
+        name: Optional[str] = "eds.suicide_attempt",
+        regex: Patterns = patterns,
+        alignment_mode: str = "strict",
+        attr: str = "TEXT",
```

Comment:
why "TEXT" and not "NORM" ?

Reply:
The regex was developped using text before normalisation